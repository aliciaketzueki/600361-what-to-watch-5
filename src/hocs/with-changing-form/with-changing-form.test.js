import React, {Fragment} from "react";
import renderer from "react-test-renderer";
import withChangingForm from "./with-changing-form";
import PropTypes from "prop-types";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withChangingForm(MockComponent);

describe(`Render withChangingForm`, () => {
  it(`Should withChangingForm render correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped>
        <Fragment />
      </MockComponentWrapped>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
