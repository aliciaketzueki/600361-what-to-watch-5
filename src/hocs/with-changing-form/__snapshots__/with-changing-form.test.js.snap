// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render withChangingForm Should withChangingForm render correctly 1`] = `<div />`;
